ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -346,22 +365,65 @@ def _count_encoding(
 
         return temp_x, temp_var_names
 
+    @staticmethod
+    def _hash_encoding(
+        adata: AnnData,
+        X: Optional[np.ndarray],
+        var_names: List[str],
+        categories: List[List[str]],
+        progress: Optional[Progress] = None,
+        task=None,
+    ) -> Tuple[np.ndarray, List[str]]:
+        """Encode categorical columns using hash encoding.
+
+        Args:
+            adata: The current AnnData object
+            X: Current (encoded) X
+            var_names: Var names of current AnnData object
+            categories: The name of the categorical columns to be encoded
+
+        Returns:
+            Encoded new X and the corresponding new var names
+        """
+        transformed_all, encoded_var_names = None, []
+        for multi_columns in categories:
+            original_values = Encoder._initial_encoding(adata, multi_columns)
+
+            encoder = HashingEncoder(return_df=False).fit(original_values)
+            # maybe let user pass a n_components kwarg here (32 bit might be useful for high cardinality input)
+            encoded_var_names += [f"ehrapycat_hash_{multi_columns[0]}" for _ in range(8)]
```

Comment:
Magic number 8. Bug or feature?

Reply:
Thats the `n_component` of HashEncode algorithm. Basically how many columns will be added. Default is 8, but should add a kwarg here maybe

Reply:
If not a kwarg, I'd suggest to at least define the 8 to a variable to know what this is about.

Reply:
Scikit learn suggests for high cardinality datasets to set this one to 32, so kwarg might be the way to go here